GH Actions: CI Savings

ax3l opened the pull request:
This uses GH Action CI resources more carefully by:

- Abort running pushes in CI if an update was pushed to a PR.
- Skip all but short style checks if a PR is opened as a draft.

## Notes on Starting or Converting a Draft PR

### Convert an open PR to draft:
![convert](https://user-images.githubusercontent.com/1353258/134959925-a197c8cf-f3d2-4df2-acf9-e612cc7f447c.png)

### Open your next PR as a draft PR:
![draft_pr](https://user-images.githubusercontent.com/1353258/134959934-9f2ddd4e-b2cc-40c4-b288-e2c38913f1c2.png)

## Notes on Avoiding to Run GH Actions on Older Forks of this Repo

if you have an older fork of this repo on your account, all tests are run twice. Once on your repo (and not shown anywhere), once on our PR. (This default changed to opt-in if you created a recent fork.)

So, if you own an older fork of this repo on GitHub, you can also safe some CO2 if you like:
Go to your fork, click on the tab `Actions` -> Select workflow -> Click on `...` -> Select "Disable Workflow"

ax3l committed:
GH Actions: Abort Prior Push

Abort running pushes in CI if an update was pushed to a PR.

ax3l committed:
GH Actions: Draft Skips Most

Skip all but short style checks if a PR is opened as a draft.

franzpoeschel commented:
Great, thank you for this!
You didn't change anything for `.appveyor.yml`, right? If I look into your CI run, MSVC did compile some files, but canceled then (e.g. [here](https://ci.appveyor.com/project/ax3l/openpmd-api/builds/40970685/job/9l80e68l5fcq4202)). Do you know what's up with that?
Also, when a PR is converted from draft to ready, this doesn't trigger the CI to run again, right? But I guess there's no way around that, so we'll just have to live with that.

ax3l committed:
CI: Groups Include Branch

On pushes to mainline, if mainline contains multiple branches, these
branches cancelled each other. This should fix this.

ax3l commented:
Yes, I think AppVeyor just had a hick-up. It's not affected

> Also, when a PR is converted from draft to ready, this doesn't trigger the CI to run again, right? But I guess there's no way around that, so we'll just have to live with that.

I think it does actually. Same was as close/reopen triggers CI, too.

franzpoeschel commented:
> I think it does actually. Same was as close/reopen triggers CI, too.

Maybe I just didn't wait long enough when trying it out then. Looks good to me then :)
